samples/bluetooth/observer/src/observer.c

```diff
@@ -9,7 +9,11 @@
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/hci.h>
 
-#define NAME_LEN 30
+#if defined(CONFIG_BT_DEVICE_NAME_DYNAMIC)
+#define NAME_LEN (CONFIG_BT_DEVICE_NAME_MAX + 1U)
+#else
+#define NAME_LEN sizeof(CONFIG_BT_DEVICE_NAME)
+#endif
```

Comment:
Ditto; this change can be reverted

Reply:
The original set the NAME_LEN to a fixed nr. I prefer this way